tests/test_view_decorators.py

```diff
@@ -229,6 +235,42 @@ def test_jwt_optional_with_no_valid_jwt(app):
     assert response.get_json() == {"msg": "Not enough segments"}
 
 
+def test_custom_jwt_required(app):
+    url = "/custom_protected"
+
+    test_client = app.test_client()
+    with app.test_request_context():
+        custom_token = create_custom_token("username", token_type="custom")
+        fresh_custom_token = create_custom_token(
+            "username", token_type="custom", fresh=True
+        )
+        refresh_token = create_refresh_token("username")
+        incorrect_custom_token = create_custom_token(
+            "username", token_type="other_custom"
+        )
+
+    # Access and fresh access should be able to access this
+    for token in (custom_token, fresh_custom_token):
+        response = test_client.get(url, headers=make_headers(token))
+        assert response.status_code == 200
+        assert response.get_json() == {"foo": "bar"}
+
+    # Test accessing jwt_required with no jwt in the request
+    response = test_client.get(url, headers=None)
+    assert response.status_code == 401
+    assert response.get_json() == {"msg": "Missing Authorization Header"}
+
+    # Test refresh token access to jwt_required
+    response = test_client.get(url, headers=make_headers(refresh_token))
+    assert response.status_code == 422
+    assert response.get_json() == {"msg": "Token of type refresh is not allowed"}
```

Comment:
Not a big deal, but I wonder if the error message should say what token type is allowed instead of this specific token is now allowed? Might make for more discoverable errors?